src/pages/workspaces/workspace/analysis/tool-utils.ts

```diff
@@ -0,0 +1,101 @@
+import _ from 'lodash/fp'
+import { isCromwellAppVisible } from 'src/libs/config'
+import * as Utils from 'src/libs/utils'
+import { getExtension } from 'src/pages/workspaces/workspace/analysis/file-utils'
+
+
+export type ToolLabel = 'Jupyter' | 'RStudio' | 'spark' | 'Azure' | 'Galaxy' | 'terminal' | 'Cromwell'
+export const toolLabelTypes: Record<ToolLabel, ToolLabel> = {
+  Jupyter: 'Jupyter',
+  RStudio: 'RStudio',
+  terminal: 'terminal',
+  spark: 'spark',
+  Azure: 'Azure',
+  Galaxy: 'Galaxy',
+  Cromwell: 'Cromwell'
+}
```

Comment:
+1 on an offline comment that it's not clear what this buys us

same for [this](https://github.com/DataBiosphere/terra-ui/pull/3447/files#diff-79707a07ad14b98951a5cb25209dbd412a61b4f23f71c2dc29f64ce961908958R2-R5)

Reply:
in chatting with Justin, he said that checking on validity of enum-like values like this was desired.  This is a pattern I have used a few times to easily give that validity check (as in the isCloudProvider type checker you link to).  Listing these values like this both enforces completeness (Typescript will error out if the object does not have a complete set of value entries because of the exact typing here)... and it gives constants to be used in code if desired... i.e.

if (x === toolLabelTypes.Azure) { ... }

Reply:
For purposes of checking if a string is a tool label, it seems like `Record<ToolLabel, any>` would work since you only need to check if the string is a key in the object. I think I'd vote against exporting this, since it adds the question of whether tool labels should be used via literals (`'Jupyter'`) or with this enum-ish object (`toolLabelTypes.Jupyter`). Also, if we want to use an enum like object and we want to check if a string is part of that object, it seems like maybe an Enum would be a better fit for `ToolLabel` than a union of literals.